server/api/api/endpoints/model_endpoints.py

```diff
@@ -377,7 +377,10 @@ async def get_model_endpoint_monitoring_metrics(
                 _wrap_coroutine_in_list(
                     run_in_threadpool(
                         mlrun.model_monitoring.get_tsdb_connector(
-                            project=project
+                            project=project,
+                            secret_provider=server.api.crud.secrets.get_project_secret_provider(
+                                project=project
```

Comment:
Can't it be handled inside the function by default?

Reply:
but in this case we will have to check each time if we are on API or not. The best practice for handling these secrets is using the secret provider (already went through it with Saar).